lorawan-encoding/tests/lorawan.rs

```diff
@@ -611,20 +637,20 @@ fn test_join_accept_creator() {
         .set_dl_settings(0)
         .set_rx_delay(0);
 
-    assert_eq!(phy.build(&key), Ok(&phy_join_accept_payload()[..]));
+    assert_eq!(phy.build(&key, &DefaultFactory), Ok(&phy_join_accept_payload()[..]));
 }
 #[test]
 #[cfg(feature = "default-crypto")]
 fn test_join_accept_creator_short_buffer() {
     let mut buf = [0u8; 16];
-    let phy_res = JoinAcceptCreator::with_options(&mut buf[..], DefaultFactory);
+    let phy_res = JoinAcceptCreator::new(&mut buf[..]);
```

Comment:
```suggestion
    let phy_res = JoinAcceptCreator::new(&mut buf);
```

Reply:
good catch, it works with both as we take a AsMut<[u8]> generic. I had [..] at some point to have them all generic over &mut [u8] instead of &mut [u8; N]. both ways work but I should be consistent at least.